src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -314,4 +317,24 @@ public static void saveAndUpdateConfigs(
             throw ExceptionsHelper.convertToOpenSearchException(e);
         }
     }
+
+    public void filterAccountsByType(SecurityDynamicConfiguration<?> configuration, String requestedAccountType) {
```

Comment:
seems like this method removes:
- internal accounts with service flag as true
- service accounts with service flag as false

If that is the case, can we rename it as `removeInvalidAccounts`

Also can you add javadoc for this one.